tests/unit/library/test_json.py

```diff
@@ -194,6 +194,82 @@ def test_bad_number(self, bad_string, good_bytes, failure_byte, allowed_bytes, m
+            (6, {"type": "integer", "exclusiveMinimum": 5}, True),
+            (5, {"type": "integer", "exclusiveMinimum": 5}, False),
+            (5, {"type": "integer", "maximum": 5}, True),
+            (6, {"type": "integer", "maximum": 5}, False),
+            (4, {"type": "integer", "exclusiveMaximum": 5}, True),
+            (5, {"type": "integer", "exclusiveMaximum": 5}, False),
+            (5, {"type": "integer", "minimum": 5, "maximum": 10}, True),
+            (4, {"type": "integer", "minimum": 5, "maximum": 10}, False),
+            (10, {"type": "integer", "exclusiveMinimum": 5, "exclusiveMaximum": 10}, False),
+            (5, {"type": "integer", "exclusiveMinimum": 5, "exclusiveMaximum": 10}, False),
+            (7, {"type": "integer", "exclusiveMinimum": 5, "exclusiveMaximum": 10}, True),
+            # --- Number type tests ---
+            (5, {"type": "number", "minimum": 5.0}, True),
+            (5.0, {"type": "number", "minimum": 5.0}, True),
+            (4.9, {"type": "number", "minimum": 5.0}, False),
+            (5.1, {"type": "number", "exclusiveMinimum": 5.0}, True),
+            (5, {"type": "number", "exclusiveMinimum": 5.0}, False),
+            (5.0, {"type": "number", "exclusiveMinimum": 5.0}, False),
+            (5, {"type": "number", "maximum": 5.0}, True),
+            (5.0, {"type": "number", "maximum": 5.0}, True),
+            (5.1, {"type": "number", "maximum": 5.0}, False),
+            (4.9, {"type": "number", "exclusiveMaximum": 5.0}, True),
+            (5, {"type": "number", "exclusiveMaximum": 5.0}, False),
+            (5.0, {"type": "number", "exclusiveMaximum": 5.0}, False),
+            (7.5, {"type": "number", "minimum": 5.0, "maximum": 10.0}, True),
+            (4.9, {"type": "number", "minimum": 5.0, "maximum": 10.0}, False),
+            (5.1, {"type": "number", "exclusiveMinimum": 5.0, "exclusiveMaximum": 10.0}, True),
+            (5.0, {"type": "number", "exclusiveMinimum": 5.0, "exclusiveMaximum": 10.0}, False),
+            (9.9, {"type": "number", "exclusiveMinimum": 5.0, "exclusiveMaximum": 10.0}, True),
+            # --- Edge cases ---
+            (0, {"type": "integer", "minimum": 0}, True),
+            (-1, {"type": "integer", "minimum": 0}, False),
+            (-5, {"type": "number", "maximum": 0}, True),
+            (0, {"type": "number", "maximum": 0}, True),
+            (5, {"type": "integer", "minimum": 5.5}, False),
+            (6, {"type": "integer", "minimum": 5.5}, True),
+            (5, {"type": "integer", "exclusiveMinimum": 5.5}, False),
+            (6, {"type": "integer", "exclusiveMinimum": 5.5}, True),
+            (5, {"type": "integer", "maximum": 5.5}, True),
+            (6, {"type": "integer", "maximum": 5.5}, False),
+            (5, {"type": "integer", "exclusiveMaximum": 5.5}, True),
+            (6, {"type": "integer", "exclusiveMaximum": 5.5}, False),
+            # --- Large numbers ---
```

Comment:
We could add a bunch more large number and precision cases (as I think manically about how floating point can really Mess You Up), but I wouldn't worry right now.